xarray/tests/test_interp.py

```diff
@@ -287,18 +325,90 @@ def test_interpolate_nd(case: int) -> None:
             "z": da["z"],
             "y": ydest,
             "x": ("y", xdest.values),
-            "x2": da["x2"].interp(x=xdest),
+            "x2": da["x2"].interp(x=xdest, method=method),
         },
     )
     assert_allclose(actual.transpose("y", "z"), expected)
 
     # reversed order
-    actual = da.interp(y=ydest, x=xdest, method="linear")
+    actual = da.interp(y=ydest, x=xdest, method=method)
     assert_allclose(actual.transpose("y", "z"), expected)
 
 
 @requires_scipy
-def test_interpolate_nd_nd() -> None:
+@pytest.mark.parametrize("method", ("cubic", "quintic"))
```

Comment:
Is it possible to test all methods supported by interpn here? “linear”, “nearest”, “slinear”, “cubic”, “quintic”, “pchip”

Basically the "separation" is an implementation detail. We just want to be sure we get correct answers regardless of that detail.